DuggaSys/microservices/shared_microservices/createNewCodeExample_ms.php

```diff
@@ -18,85 +11,24 @@
 session_start();
 
 // Global variables
-$opt = getOP('opt');
-$link = getOP('link');
 $courseid = getOP('courseid');
-$coursevers = getOP('coursevers');
+$examplename=getOP('examplename');
 $sectname = getOP('sectname');
-$kind = getOP('kind');
-$comments = getOP('comments');
-$visibility = getOP('visibility');
-$highscoremode = getOP('highscoremode');
-$pos = getOP('pos');
-$grptype = getOP('grptype');
-$gradesys=getOP('gradesys');
-$tabs=getOP('tabs');
-
-getUid(); // call to microservice getUid_ms.php
-$userid = $_SESSION['uid'];
-
-if (strcmp($opt, "NEW") === 0) {
-
-	// Insert a new code example and update variables accordingly.
-	if ($link == -1) {
-		$query1 = $pdo->prepare("SELECT * FROM codeexample ORDER BY exampleid DESC LIMIT 1");
-		if (!$query1->execute()) {
-			$error = $query1->errorInfo();
-			$debug = "Error reading entries" . $error[2];
-		}
-
-		foreach ($query1->fetchAll() as $row) {
-			$sname = $row['exampleid'] + 1;
-		}
-
-		$sname = $sectname . $sname;
-		$query2 = $pdo->prepare("INSERT INTO codeexample(cid,examplename,sectionname,uid,cversion) values (:cid,:ename,:sname,1,:cversion);");
-
-		$query2->bindParam(':cid', $courseid);
-		$query2->bindParam(':cversion', $coursevers);
-		$query2->bindParam(':ename', $sectname);
-		$query2->bindParam(':sname', $sname);
-
-		if (!$query2->execute()) {
-			$error = $query2->errorInfo();
-			$debug = "Error updating entries" . $error[2];
-		}
-
-		$link = $pdo->lastInsertId();
-	}
-
-	$query = $pdo->prepare("INSERT INTO listentries (cid,vers, entryname, link, kind, pos, visible,creator,comments, gradesystem, highscoremode, groupKind) 
-												 VALUES(:cid,:cvs,:entryname,:link,:kind,:pos,:visible,:usrid,:comment, :gradesys, :highscoremode, :groupkind)");
+$coursevers = getOP('coursevers');
+$userid = getUid();
```

Comment:
I think this is fine, but i notice that the global variables probably would already be harvested in the file that includes this file. This will serve as additional safety.